powerauth-java-cmd-lib/src/main/java/io/getlime/security/powerauth/lib/cmd/steps/v3/TokenAndEncryptStep.java

```diff
@@ -0,0 +1,172 @@
+import io.getlime.security.powerauth.lib.cmd.util.VerifySignatureUtil;
+import io.getlime.security.powerauth.rest.api.model.response.v3.EciesEncryptedResponse;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpMethod;
+import org.springframework.stereotype.Component;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+
+/**
+ * Token and encrypt step generates token authentication for request data and performs encryption using ECIES encryption in activation scope.
+ *
+ * <p><b>PowerAuth protocol versions:</b>
+ * <ul>
+ *     <li>3.0</li>
+ *     <li>3.1</li>
+ * </ul>
+ * <p>
+ *  @author Roman Strobl, roman.strobl@wultra.com
+ */
+@Component
+public class TokenAndEncryptStep extends AbstractBaseStep<TokenAndEncryptStepModel, EciesEncryptedResponse> {
+
+    private final PowerAuthHeaderFactory powerAuthHeaderFactory;
+
+    /**
+     * Constructor.
+     * @param powerAuthHeaderFactory PowerAuth header factory.
+     * @param resultStatusService Result status service.
+     * @param stepLoggerFactory Step logger factory.
+     */
+    @Autowired
+    public TokenAndEncryptStep(
+            PowerAuthHeaderFactory powerAuthHeaderFactory,
+            ResultStatusService resultStatusService,
+            StepLoggerFactory stepLoggerFactory) {
+        super(PowerAuthStep.TOKEN_ENCRYPT, PowerAuthVersion.VERSION_3, resultStatusService, stepLoggerFactory);
+
+        this.powerAuthHeaderFactory = powerAuthHeaderFactory;
+    }
+
+    /**
+     * Constructor for backward compatibility
+     */
+    public TokenAndEncryptStep() {
+        this(
+                BackwardCompatibilityConst.POWER_AUTH_HEADER_FACTORY,
+                BackwardCompatibilityConst.RESULT_STATUS_SERVICE,
+                BackwardCompatibilityConst.STEP_LOGGER_FACTORY
+        );
+    }
+
+    @Override
+    protected ParameterizedTypeReference<EciesEncryptedResponse> getResponseTypeReference() {
+        return PowerAuthConst.RESPONSE_TYPE_REFERENCE_V3;
+    }
+
+    @Override
+    public StepContext<TokenAndEncryptStepModel, EciesEncryptedResponse> prepareStepContext(StepLogger stepLogger, Map<String, Object> context) throws Exception {
+        TokenAndEncryptStepModel model = new TokenAndEncryptStepModel();
+        model.fromMap(context);
+
+        RequestContext requestContext = RequestContext.builder()
+                .signatureHttpMethod(model.getHttpMethod())
+                .uri(model.getUriString())
+                .build();
+
+        StepContext<TokenAndEncryptStepModel, EciesEncryptedResponse> stepContext =
+                buildStepContext(stepLogger, model, requestContext);
+
+        // Verify that HTTP method is set
+        if (model.getHttpMethod() == null) {
+            stepLogger.writeError("token-encrypt-error-http-method", "HTTP method not specified", "Specify HTTP method to use for sending request");
```

Comment:
let's use `getStep().id() + ` instead of `token-encrypt`

Reply:
Sure.